File: GCD/greatestCommonDivisor.cpp
#include <bits/stdc++.h>
using namespace std;
int greatestCommonDivisor(int a, int b) {
	// Recursive function to find GCD of a and b
	if (b == 0)
		return a;
	return greatestCommonDivisor(b, a % b);
}
int main() {
	int a = 10, b = 6;
	cout << greatestCommonDivisor(a, b) << endl;
	return 0;
}

File: 26_GCD/greatestCommonDivisor.py
# Compute GCD of A and B
def greatestCommonDivisor(N, M):
    if M == 0 :
        return N
    return greatestCommonDivisor(M, N%M)
    pass


if __name__ == "__main__":

    # Input
    array = [1, 2, 1, 3, 5, 6, 4]
    N = 10
    M = 6

    print(greatestCommonDivisor(N, M))
